lib/web_bruteforcer.py

```diff
@@ -0,0 +1,68 @@
+# Web Content Bruteforcer library (Part of the PixelToolkit project)
+
+from lib.utils import cli_error, log, LogUrgency, cli_print, MessageType, Color
+import requests
+import threading
+
+DEFAULT_WORDLIST = "lib/wordlist.txt"
+
+
+class WebBruteforcer():
+    def __init__(self, cli=False):
+        self.cli = cli
+        self.results = []
+
+    def scan(self, url: str, wordlist: str, n_threads: int):
+        log(f"Starting web bruteforcer. URL={url} threads={n_threads}", LogUrgency.INFO)
+        # TODO: support multiple FUZZ parameters in the URL
+        if "FUZZ" not in url:
+            if self.cli:
+                cli_error("You need to specify the fuzzing point using the 'FUZZ' keyword")
+            return False
+        wordlist = open(wordlist or DEFAULT_WORDLIST).read().strip().split()
+
+        threads = []
+        for i in range(n_threads):
+            thread = threading.Thread(target=self.scanning_thread, args=(url, wordlist, i, n_threads))
+            threads.append(thread)
+            thread.start()
+
+        for thread in threads:
+            thread.join()
+
+        if self.cli:
+            cli_print("Web content bruteforce finished", MessageType.INFO)
+
+        return self.results
+
+    # I *think* python will pass the wordlist by reference. This is quite important as this wordlist object
+    # can be multiple hundreds of Megabytes or even Gigabytes in size and cloning it for each thread is not
+    # ideal to say the least.
+
+    def scanning_thread(self, url: str, wordlist: list, starting_index: int, n_threads: int):
```

Comment:
I'm not sure if that `list` type is a correct way of expressint a list of strings. there is also `[str]` but Python doesn't seem to care about those type hints so when I get to setting up a type checker then we will know for sure